client/web/src/user/profile/UserProfile.tsx

```diff
@@ -0,0 +1,52 @@
+import { formatDistanceToNowStrict } from 'date-fns'
+import React from 'react'
+import { UserAreaRouteContext } from '../area/UserArea'
+
+export const UserProfile: React.FunctionComponent<
+    Pick<UserAreaRouteContext, 'user'> & { isSourcegraphDotCom: boolean }
+> = ({ user, isSourcegraphDotCom }) => {
+    const primaryEmail = user.emails.find(email => email.isPrimary)?.email
+
+    const userData: {
+        name: string
+        value: string
+        condition?: boolean
+    }[] = [
+        {
+            name: 'Username',
+            value: user.username,
+        },
+        {
+            name: 'Display name',
+            value: user.displayName || 'Not set',
+            condition: !!user.displayName,
+        },
+        {
+            name: 'User since',
+            value: formatDistanceToNowStrict(new Date(user.createdAt), { addSuffix: true }),
+        },
+        {
+            name: 'Email',
+            value: primaryEmail || 'Not set',
+            condition: !!primaryEmail && !isSourcegraphDotCom, // Don't show email on Sourcegraph.com
+        },
+        {
+            name: 'Site admin',
+            value: user.siteAdmin ? 'Yes' : 'No',
+            condition: user.siteAdmin && !isSourcegraphDotCom, // Don't show site admin status on Sourcegraph.com
+        },
+    ]
+
+    return (
+        <dl>
+            {userData.map(data =>
+                data.condition !== false ? (
```

Comment:
nit: since the array assigned to `userData` is controlled above, perhaps it would simplify the `map` method if `condition` was required and specified in every element with `true` value?

```suggestion
                data.condition ? (
```

Reply:
I tried both and I'm 50/50 on which one looks nicer :P so I'll switch to what you suggest.

Reply:
Yep, same here. No strong preference. Perhaps I should have kept this comment to myself 🤭